### ISSUE TITLE
MultioutputWrapper does not reset cleanly

### ISSUE THREAD

**Issue by phschoepf** [Lightning-AI/metrics#1436]
## 🐛 Bug

Calling `MultioutputWrapper.compute()` after `MultioutputWrapper.reset()` returns old metrics that should have been cleared by the reset. 

#### Code sample

```py
import torch
import torchmetrics

base_metric = torchmetrics.ConfusionMatrix(task="multiclass", num_classes=2)
cf = torchmetrics.MultioutputWrapper(base_metric, num_outputs=2)

cf(torch.tensor([[0,0]]), torch.tensor([[0,0]]))
print("First result: ", cf.compute())

cf.reset()

cf(torch.tensor([[1,1]]), torch.tensor([[0,0]]))
print("Second result: ", cf.compute())
```

Output:
```
First result:  [tensor([[1, 0], [0, 0]]), tensor([[1, 0], [0, 0]])]
Second result:  [tensor([[1, 0], [0, 0]]), tensor([[1, 0], [0, 0]])]
```
The old output is returned even after resetting and entering new data. If the fist metric computation is omitted, the second metric is as expected.

Importantly, this bug only occurs when using `forward()` to enter data, while `update()` works as expected.

### Expected behavior

The result of the second computation should be independent of the first. Furthermore, forward and update should produce the same state as specified in the docs.

### Environment

- torchmetrics 0.10.3, installed from pypi
- Python 3.8.9

### Attempts to fix

Adding `super().reset()` (as done in e.g. the minmax wrapper) at the top of the reset method seems to fix the bug. 
https://github.com/Lightning-AI/metrics/blob/7b505ff1a3b88181bef2b0cdfa21ec593dcda3ff/src/torchmetrics/wrappers/multioutput.py#L133
